### ISSUE TITLE
S3 Select "OverflowError: Python int too large to convert to C long"

### ISSUE THREAD

**Issue by kukushking** [aws/aws-sdk-pandas#2236]
### Describe the bug

```
---------------------------------------------------------------------------
OverflowError                             Traceback (most recent call last)
Cell In[8], line 1
----> 1 df = wr.s3.select_query(
      2                               sql='SELECT * FROM s3object s where s.\"id\" >= 5',
      3                               path ="s3://.../prefix/",
      4                               input_serialization = "Parquet",
      5                               input_serialization_params = {},
      6                               use_threads=True
      7                              )

File /usr/local/lib/python3.8/site-packages/awswrangler/_utils.py:167, in validate_kwargs.<locals>.decorator.<locals>.inner(*args, **kwargs)
    164 if condition_fn() and len(passed_unsupported_kwargs) > 0:
    165     raise exceptions.InvalidArgument(f"{message} `{', '.join(passed_unsupported_kwargs)}`.")
--> 167 return func(*args, **kwargs)

File /usr/local/lib/python3.8/site-packages/awswrangler/s3/_select.py:289, in select_query(sql, path, input_serialization, input_serialization_params, compression, scan_range_chunk_size, path_suffix, path_ignore_suffix, ignore_empty, use_threads, last_modified_begin, last_modified_end, boto3_session, s3_additional_kwargs, pyarrow_additional_kwargs)
    286 arrow_kwargs = _data_types.pyarrow2pandas_defaults(use_threads=use_threads, kwargs=pyarrow_additional_kwargs)
    287 executor: _BaseExecutor = _get_executor(use_threads=use_threads)
    288 tables = list(
--> 289     itertools.chain(*ray_get([_select_query(path=path, executor=executor, **select_kwargs) for path in paths]))
    290 )
    291 return _utils.table_refs_to_df(tables, kwargs=arrow_kwargs)

File /usr/local/lib/python3.8/site-packages/awswrangler/s3/_select.py:289, in <listcomp>(.0)
    286 arrow_kwargs = _data_types.pyarrow2pandas_defaults(use_threads=use_threads, kwargs=pyarrow_additional_kwargs)
    287 executor: _BaseExecutor = _get_executor(use_threads=use_threads)
    288 tables = list(
--> 289     itertools.chain(*ray_get([_select_query(path=path, executor=executor, **select_kwargs) for path in paths]))
    290 )
    291 return _utils.table_refs_to_df(tables, kwargs=arrow_kwargs)

File /usr/local/lib/python3.8/site-packages/awswrangler/_distributed.py:102, in Engine.dispatch_on_engine.<locals>.wrapper(*args, **kw)
    100 @wraps(func)
    101 def wrapper(*args: Any, **kw: Dict[str, Any]) -> Any:
--> 102     return cls.dispatch_func(func)(*args, **kw)

File /usr/local/lib/python3.8/site-packages/awswrangler/s3/_select.py:137, in _select_query(path, executor, sql, input_serialization, input_serialization_params, compression, scan_range_chunk_size, boto3_session, s3_additional_kwargs)
    127 if any(
    128     [
    129         compression,
   (...)
    133 ):  # Scan range is only supported for uncompressed CSV/JSON, CSV (without quoted delimiters)
    134     # and JSON objects (in LINES mode only)
    135     scan_ranges = [None]  # type: ignore[assignment]
--> 137 return executor.map(_select_object_content, s3_client, itertools.repeat(args), scan_ranges)

File /usr/local/lib/python3.8/site-packages/awswrangler/_executor.py:51, in _ThreadPoolExecutor.map(self, func, boto3_client, *args)
     49 if self._exec is not None:
     50     iterables = (itertools.repeat(boto3_client), *args)
---> 51     return list(self._exec.map(func, *iterables))
     52 # Single-threaded
     53 return list(map(func, *(itertools.repeat(boto3_client), *args)))

File /usr/local/lib/python3.8/concurrent/futures/_base.py:619, in Executor.map.<locals>.result_iterator()
    616 while fs:
    617     # Careful not to keep a reference to the popped future
    618     if timeout is None:
--> 619         yield fs.pop().result()
    620     else:
    621         yield fs.pop().result(end_time - time.monotonic())

File /usr/local/lib/python3.8/concurrent/futures/_base.py:444, in Future.result(self, timeout)
    442     raise CancelledError()
    443 elif self._state == FINISHED:
--> 444     return self.__get_result()
    445 else:
    446     raise TimeoutError()

File /usr/local/lib/python3.8/concurrent/futures/_base.py:389, in Future.__get_result(self)
    387 if self._exception:
    388     try:
--> 389         raise self._exception
    390     finally:
    391         # Break a reference cycle with the exception in self._exception
    392         self = None

File /usr/local/lib/python3.8/concurrent/futures/thread.py:57, in _WorkItem.run(self)
     54     return
     56 try:
---> 57     result = self.fn(*self.args, **self.kwargs)
     58 except BaseException as exc:
     59     self.future.set_exception(exc)

File /usr/local/lib/python3.8/site-packages/awswrangler/_distributed.py:102, in Engine.dispatch_on_engine.<locals>.wrapper(*args, **kw)
    100 @wraps(func)
    101 def wrapper(*args: Any, **kw: Dict[str, Any]) -> Any:
--> 102     return cls.dispatch_func(func)(*args, **kw)

File /usr/local/lib/python3.8/site-packages/awswrangler/_utils.py:804, in try_it(f, ex, ex_code, base, max_num_tries, *args, **kwargs)
    802 for i in range(max_num_tries):
    803     try:
--> 804         return f(*args, **kwargs)
    805     except ex as exception:
    806         if ex_code is not None and hasattr(exception, "response"):

File /usr/local/lib/python3.8/site-packages/awswrangler/s3/_select.py:82, in _select_object_content(s3_client, args, scan_range)
     77 if not request_complete:
     78     raise exceptions.S3SelectRequestIncomplete(
     79         f"S3 Select request for path {args['Key']} is incomplete as End Event was not received"
     80     )
---> 82 return _utils.list_to_arrow_table(mapping=payload_records)

File /usr/local/lib/python3.8/site-packages/awswrangler/_utils.py:905, in list_to_arrow_table(mapping, schema, metadata)
    903         v = [row[n] if n in row else None for row in mapping]
    904         arrays.append(v)
--> 905     return pa.Table.from_arrays(arrays, names, metadata=metadata)
    906 for n in schema.names:
    907     v = [row[n] if n in row else None for row in mapping]

File /usr/local/lib/python3.8/site-packages/pyarrow/table.pxi:3657, in pyarrow.lib.Table.from_arrays()

File /usr/local/lib/python3.8/site-packages/pyarrow/table.pxi:1406, in pyarrow.lib._sanitize_arrays()

File /usr/local/lib/python3.8/site-packages/pyarrow/table.pxi:1387, in pyarrow.lib._schema_from_arrays()

File /usr/local/lib/python3.8/site-packages/pyarrow/array.pxi:320, in pyarrow.lib.array()

File /usr/local/lib/python3.8/site-packages/pyarrow/array.pxi:39, in pyarrow.lib._sequence_to_array()

File /usr/local/lib/python3.8/site-packages/pyarrow/error.pxi:144, in pyarrow.lib.pyarrow_internal_check_status()

OverflowError: Python int too large to convert to C long
```

### How to Reproduce

```
import awswrangler as wr
import pandas as pd

path = "s3://..."

df = pd.DataFrame(
        {
            # The values below overflow pa.int64()
            "c0": [9223372036854775807, 9223372036854775808, 9223372036854775809],
            "c1": ["foo", "boo", "bar"],
        }
    )
  wr.s3.to_parquet(
      df,
      path,
      compression="snappy",
  )
  df2 = wr.s3.select_query(
      sql="select * from s3object",
      path=path,
      input_serialization="Parquet",
      input_serialization_params={},
      use_threads=False,
  )
```

### OS

OSX / Unix

### Python version

3.10

### AWS SDK for pandas version

3.0.0

**Comment by kukushking** [comment]
Pyarrow issue ref https://github.com/apache/arrow/issues/32664

**Comment by jaidisido** [comment]
Fixed in #2237, thanks Anton
